packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,553 @@
+## Status
+
+- **Last Updated:** August 3rd 2021
+- **Stable:** Yes
+- **Risk of Breaking Change:** Medium
+- **Compatibility:** `>= @onflow/fcl@0.0.77`
+
+## Definitions
+
+This document is written with the perspective that _you_, who is reading this is an FCL Wallet Developer. All references to _you_ in this doc are done with this perspective in mind.
+
+# Overview
+
+Flow Client Library (FCL) approaches the idea of blockchain wallets on Flow in a different way than how wallets may be supported on other blockchains. For example, with FCL, a wallet is not necessarily limited to being a browser extention or even a native application on a users device. FCL offers wallet developers the flexibility and freedom to build many different types of applications. Since wallet applications can take on many forms, we needed to create a way for these varying applications to be able to communicate and work together.
+
+FCL acts in many ways as a protocol to facilitate communication and configuration between the different parties involved in a blockchain application. An _Application_ can use FCL to _authenticate_ users, and request _authorizations_ for transactions, as well as mutate and query the _Blockchain_. An application using FCL offers it's _Users_ a way to connect and select any number of Wallets Providers and their Wallet Services. An selected _Wallet_ provides an Applications instance of FCL with configuration information about itself and its Wallet Services, allowing the _User_ and _Application_ to interact with them.
+
+In the following paragraphs we'll explore ways in which you, as a wallet developer, can integrate with FCL by providing implementataions of various FCL services. 
+
+The following services will be covered:
+- Authentication (Authn) Service
+- Authorization (Authz) Service
+- User Signature Service
+- Pre-Authz Service
+
+# Authentication
```

Comment:
we could move this under the service methods, for each service specify what we are expecting it to return and then say something like `while you can use any of the above service methods for this service, the following example uses the _____ method` and then give an example of using it. each service could use a different service method as part of its example.